src/vars.c

```diff
@@ -852,6 +847,49 @@ static Obj EvalElmListLevel(Expr expr)
 }
 
 
+/****************************************************************************
+**
+*F  EvalElmMatLevel(<expr>)  . . . . . .  select elements of several matrices
+**
+**  'EvalElmMatLevel' evaluates the matrix element  expression <expr> of  the
+**  form '<list>...{<positions>}...[<row>,<col>]',  where there  may actually
+**  be several '{<positions>}' selections between <list> and  '[<position>]'.
+**  The  number of those is called   the level.  'EvalElmMatLevel'  goes that
+**  deep  into the left operand  and  selects the element at <row>,<col> from
+**  each of those matrices. For example,  if the level is 1, the left operand
+**  must be a list of matrices and 'EvalElmMatLevel'  selects the element  at
+**  <row>,<col>  from each  of  the matrices  and returns  the list of  those
+**  values.
+*/
+static Obj EvalElmMatLevel(Expr expr)
+{
+    Obj                 matrices;       // matrices
+    Obj                 row;            // row position
+    Obj                 col;            // column position
+    UInt                level;          // level
+    Obj                 ixs;
+
+    // evaluate matrices (if this works, then <matrices> is nested <level>
+    // deep, checking it is nested <level>+1 deep is done by 'ElmListLevel')
+    matrices = EVAL_EXPR(READ_EXPR(expr, 0));
+    row = EVAL_EXPR(READ_EXPR(expr, 1));
+    col = EVAL_EXPR(READ_EXPR(expr, 2));
+    level = READ_EXPR(expr, 3);
+
+    ixs = NEW_PLIST(T_PLIST, 2);
+    SET_ELM_PLIST(ixs, 1, row);
+    SET_ELM_PLIST(ixs, 2, col);
+    CHANGED_BAG(ixs);
+    SET_LEN_PLIST(ixs, 2);
+
+    // select the elements from several matrices (store them in <matrices>)
+    ElmListLevel( matrices, ixs, level );
```

Comment:
```suggestion
    ElmListLevel(matrices, ixs, level);
```
Maybe you can run `git clang-format` on this patch?

Reply:
> Maybe you can run `git clang-format` on this patch?

Ah, nice, I did not know this command :-) Thanks for the hint!

I have updated the line together with the line where I copied this from. `git clang-format` also wanted to remove whitespace in variable declarations at the top of the function. I have not included this change because it would make the new function inconsistent with all other functions in the file.